apis/project.cattle.io/v3/app_types.go

```diff
@@ -75,10 +76,12 @@ type AppRevisionStatus struct {
 }
 
 type AppUpgradeConfig struct {
-	ExternalID string            `json:"externalId,omitempty"`
-	Answers    map[string]string `json:"answers,omitempty"`
+	ExternalID   string            `json:"externalId,omitempty"`
+	Answers      map[string]string `json:"answers,omitempty"`
+	ForceUpgrade bool              `json:"forceUpgrade,omitempty"`
 }
 
 type RollbackRevision struct {
 	RevisionName string `json:"revisionName,omitempty" norman:"type=reference[/v3/project/schemas/apprevision]"`
+	ForceUpgrade bool   `json:"forceUpgrade,omitempty"`
```

Comment:
why does rollbackRevision need this field?

Reply:
@cjellick ForceUpgrade can be added to rollback in case a force rollback that needs to be happened.  @guangbochen seems like you are not adding this field to rancher. https://github.com/rancher/rancher/pull/17193/commits/9da309e3ebb99af65ed5d16332b2adaf710cd917#diff-6ed5c512e37e076a2ed0c2c25246a54dL112